File: appium-android-9-tryout/tests/javascript/helloworld.js
const fs = require('fs')
const path = require('path')
const wdio = require("webdriverio");

const TEST_DIR=__dirname
const SCREENCAPTURE_DIR=path.resolve(`${TEST_DIR}/screen`)

const opts = {
  path: '/wd/hub',
  port: 4723,
  capabilities: {
    platformName: "Android",
    platformVersion: "9.0",
    app: "/root/apk_pool/ApiDemos-debug.apk",
    // fullReset: true
    // deviceName: "nexus_5_7.1.1",
    // appPackage: "io.appium.android.apis",
    // appActivity: ".view.TextFields",
    // automationName: "UiAutomator2"
  }
};

async function main () {
  const client = await wdio.remote(opts);

  let screenshot = await client.takeScreenshot();
  fs.writeFileSync(`${SCREENCAPTURE_DIR}/capture_using_javascript.png`,screenshot,{encoding:'base64'})

  await client.deleteSession();
}

main();

File: appium-wd-helloworld-tryout/tests/python/helloworld.py
import os,sys
from pprint import pprint

from appium import webdriver

TEST_DIR=os.path.abspath(os.path.dirname(__file__))
SCREENCAPTURE_DIR=os.path.abspath('{}/screen'.format(TEST_DIR))

desired_caps = {}
desired_caps['platformName'] = 'Android'
# desired_caps['appPackage'] = 'net.openwritings.xmtl'
# desired_caps['appActivity'] = '.MainActivity'
desired_caps['deviceName']='nexus_5_8'

# reference to server side
desired_caps['app'] = '/root/apk_pool/ApiDemos-debug.apk'

# This will launch your Android application.
driver = webdriver.Remote('http://localhost:4444/wd/hub', desired_caps)

search_box_element = driver.find_element_by_xpath('/hierarchy/android.widget.FrameLayout/android.view.ViewGroup/android.widget.FrameLayout[1]/android.view.ViewGroup/android.widget.TextView')

# reference to machine running the script
driver.save_screenshot('{}/capture_using_python.png'.format(SCREENCAPTURE_DIR))

print(search_box_element)
